### PR TITLE
Scan: Disable delegate button when balance is insufficient

### LINKED ISSUE #2986: Disable delegate, undelegate and redelegate button when balance is insufficient

### PR DESCRIPTION
**Pull request by Pitchanai**
Fixed: #2986 

## Implementation details
- Disable delegate, undelegate and redelegate buttun when balance is insufficient
- Fix button height not match
- Refactor few codes

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Review by pzshine — APPROVED** [review]

File: scan/src/components/modal/ValidatorStakingInfo.re [hunk 1]
```diff
@@ -25,22 +25,26 @@ module Styles = {
...
-    switch (validatorInfoSub) {
-    | Data(validatorInfo) =>
+    switch (allSub) {
+    | Data((validatorInfo, {amount: {amount}})) =>
+      let disabled = amount === 0.;
       <div className={CssHelper.flexBox()} id="validatorDelegationinfoDlegate">
         <Button
           px=20
           py=5
+          disabled
```
Comment (evilpeach) [inline scan/src/components/modal/ValidatorStakingInfo.re]: Is it correct? Should it be disabled when user's balance is not zero?

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/components/modal/ValidatorStakingInfo.re [hunk 2]
```diff
@@ -52,14 +58,16 @@ module ButtonSection = {
           <Text value="Delegate" weight=Text.Medium nowrap=true block=true />
         </Button>
         <HSpacing size=Spacing.md />
-        <Button px=20 py=5 variant=Button.Outline onClick={_ => {undelegate()}}>
+        <Button
+          px=20 py=5 variant=Button.Outline disabled=disableDelegate onClick={_ => {undelegate()}}>
```
Comment (evilpeach) [inline scan/src/components/modal/ValidatorStakingInfo.re]: remove `{}` Let's change to `onClick={_ => undelegate()}`

File: scan/src/components/modal/ValidatorStakingInfo.re [hunk 3]
```diff
@@ -52,14 +58,16 @@ module ButtonSection = {
...
           <Text value="Undelegate" weight=Text.Medium nowrap=true block=true />
         </Button>
         <HSpacing size=Spacing.md />
-        <Button px=20 py=5 variant=Button.Outline onClick={_ => {redelegate()}}>
+        <Button
+          px=20 py=5 variant=Button.Outline disabled=disableDelegate onClick={_ => {redelegate()}}>
```
Comment (evilpeach) [inline scan/src/components/modal/ValidatorStakingInfo.re]: here as well

File: scan/src/components/modal/ValidatorStakingInfo.re [hunk 4]
```diff
@@ -25,22 +25,28 @@ module Styles = {
...
     let undelegate = () =>
       validatorAddress->SubmitMsg.Undelegate->SubmitTx->OpenModal->dispatchModal;
     let redelegate = () =>
       validatorAddress->SubmitMsg.Redelegate->SubmitTx->OpenModal->dispatchModal;
 
-    switch (validatorInfoSub) {
-    | Data(validatorInfo) =>
+    switch (allSub) {
+    | Data((validatorInfo, {balance}, {amount: {amount}})) =>
+      let disabled = balance |> Coin.getBandAmountFromCoins == 0.;
```
Comment (evilpeach) [inline scan/src/components/modal/ValidatorStakingInfo.re]: `disabledNoBalance`

File: scan/src/components/modal/ValidatorStakingInfo.re [hunk 5]
```diff
@@ -25,22 +25,28 @@ module Styles = {
...
+      let disableDelegate = amount == 0.;
```
Comment (evilpeach) [inline scan/src/components/modal/ValidatorStakingInfo.re]: `disabledNoStake`

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**